modules/drivers/mongo/src/metabase/driver/mongo/database.clj

```diff
@@ -27,11 +27,8 @@
 (defn- update-ssl-db-details
   [db-details]
   (-> db-details
-      (assoc :client-ssl-key (secret/get-secret-string db-details "client-ssl-key"))
-      (dissoc :client-ssl-key-creator-id
-              :client-ssl-key-created-at
-              :client-ssl-key-id
-              :client-ssl-key-source)))
+      (secret/clean-secret-properties-from-details :mongo)
+      (assoc :client-ssl-key (secret/value-as-string :mongo db-details "client-ssl-key"))))
```

Comment:
The following comment is not specific to this backport. I was thinking if this function (and other similar ones) should get the driver passed in so that inheriting drivers can customize them by implementing `secret/value-as-string`.